openid.py

```diff
@@ -28,26 +28,78 @@
 
 import base64
 import binascii
+import datetime
 import json
 import urllib.request
 
 from cryptography.hazmat.backends import default_backend
 from cryptography.hazmat.primitives.asymmetric import rsa
+from datetime import (datetime, timedelta)
 
 class OpenIDConnectConfiguration(object):
-    def __init__(self, url):
-        self.load_config(url)
+    """
+    Represents an OpenID configuration.
+    """
     
-    def load_config(self, url):
+    lastUpdated = datetime.now()
+    signing_keys = {}
+
+    def __init__(self, url):
+        """
+        Construtor.
+        """
+        self._url = url
+        keys = OpenIDConnectConfiguration.signing_keys.get(url)
+        if keys is None:
+            OpenIDConnectConfiguration.refresh_cache(url)
+
+    def get_key(self, keyId):
+        """
+        Gets the public key from the cache given the key ID.
+        """
+        diff = datetime.now() - OpenIDConnectConfiguration.lastUpdated
+
+        # Refresh the cache if it's more than 5 days old.
+        if diff.total_seconds() > 5 * 24 * 60 * 60:
```

Comment:
Consider making the logic here consistent with the logic used in the Java sample. Does the node.js sample have this "refresh" logic? I couldn't find it.